provisioning_templates/snippet/set_hostname.erb

```diff
@@ -0,0 +1,20 @@
+<%#
+name: set_hostname
+snippet: true
+model: ProvisioningTemplate
+kind: snippet
+-%>
+if [ `command -v hostnamectl` ]; then
+  hostnamectl set-hostname <%= @hostname %>
+else
+  hostname <%= @hostname %>
+  # persist the change to survive  reboot
+  echo "<%= @hostname -%>" > /etc/hostname
+fi
+
+if grep '^127.0.0.1\s*<%= @hostname -%>$' /etc/hosts; then
+  echo "skipping /etc/hosts update, record is already present"
+else
+  echo "adding record to the first line of /etc/hosts"
+  sed -i '1s/^/127.0.0.1   <%= @hostname -%>\n/' /etc/hosts
```

Comment:
@mmoll what if I prepared my image with custom hosts file with some specific records? sed should be safe enough... I'm not sure about the localhost, but I suppose it does not hurt to change it to
`sed -i '1s/^/127.0.0.1   <%= @hostname -%> localhost\n/' /etc/hosts`

would that be acceptable?